AdblockPlusSafariExtension/AdblockPlus+Parsing.m

```diff
@@ -0,0 +1,308 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-2015 Eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/&gt.
+ */
+
+#import "AdblockPlus+Parsing.h"
+
+// yajl is sax-like json parser. Content blocker extension has limited amount of memory,
+// so that it is not possible to load whole filter list at once.
+#include <yajl/yajl_parse.h>
```

Comment:
We need to append whitelist rules to existing filter lists file. I was trying to force iOS to somehow process multiple files with filter list at once without any success. Only working option is to recreate file with append whitelist rules.

I tried to process with default iOS tools on extension side. However, content blocker extension has limited amount of memory to run and the extension was crashing with default iOS parser (it run out of memory), so that I have to either move parsing to host app or find more memory efficient parser.

 I found this simple stream parser which seems to works fine.